tasks/vmaas_sync/package_sync.go

```diff
@@ -218,15 +235,62 @@ func storePackageDetailsFrmPkgListItems(tx *gorm.DB, nameIDs map[string]int, pkg
 	}
 	utils.Log("packages", len(toStore)).Info("Collected packages to store")
 
+	err := toStore.storeOrUpdate(tx)
+	return err
+}
+
+//nolint:funlen
+func (pkgs pkgSlice) storeOrUpdate(tx *gorm.DB) error {
```

Comment:
Maybe `upsert` since it is insert or update?
```suggestion
func (ps packages) upsert(tx *gorm.DB) error {
```

Reply:
Please don't rename ii to `upsert` because it would suggest that it's doing database upsert (aka insert ... on conflict update) which it does not.

Reply:
At the first glance, I got confused because it does not store **or** update either. Let's not waste time on this.